### ISSUE TITLE
os_printf does not work for a certain amount of characters.

### ISSUE THREAD

**Issue by brianleppez** [rellermeyer/course_os#31]
When attempting to print out between 52 and 60 characters, the program runs in an infinite loop.

**Comment by lkolbly** [comment]
Having trouble reproducing. Can you post a commit ID which shows the bug? (I'll probably start looking into this tomorrow)

**Comment by brianleppez** [comment]
499a5acf445c83cac07db5c75ddca6a90947212d

**Comment by lkolbly** [comment]
This bug appears to be linked in some way to the os_memset bug, which has been fixed in master. I just pushed some changes to the branch, look at streams.c:91 and streams.c:22.

This doesn't appear to do what you think it does:
struct foo *p;
memset(&p, 0, sizeof(p))
p->field = 123;

Because that's equivalent to:
struct foo *p = 0x0;
p->field = 123;
Which is clearly not what you want to do.

I think this is what is behind the bug in question.
